### ISSUE TITLE
Data to OSM

### ISSUE THREAD

**Issue by habi** [kort/kort-native#55]
Not really an issue, but a request for clarification...

I wonder about how my 'work' goes to OSM: On the website you mention that
> As soon as three players have solved the same mission with identical answers,
> its solution is then updated in OpenStreetMap

I might be in a remote/rural area or in an area where there are no other kort players.
This means that my missions will never be solved with an identical answer, e.g. the updates are never going to OSM and are essentially lost.

Can you comment on this?

**Comment by sfkeller** [comment]
It's true, that in rural areas there are probably less mappers. But I would not assume you remain alone forever... The average of version numbers of all OSM ways in Switzerland - where only 20% live in rural areas - current is 4.9. So I think chances are there that other users pass by. I surely can tell more after we have collected some statistics. Having 3 is also some compromise against organised cheating, because at least 3 users must cooperate actively.

**Comment by habi** [comment]
> Having 3 is also some compromise against organised cheating because at least 3 users must cooperate actively.

I agree for the Kort high scores, but there's no one preventing 'cheating' on OSM and it seems to work quite well there...

**Comment by matkoniecz** [comment]
I decided against using Kort because I am not interested in my edits waiting in limbo for months/years.

Maybe publish data to OSM if there is one week without confirmation/rejection and wait for 2 confirmations for purposes of scoring?

**Comment by sfkeller** [comment]
matkoniecz: Kort was designed for gaming - not OSM editing. If you want to edit OSM, I recommend using StreetComplete or Vespucci.

**Comment by sfkeller** [comment]
Having this in mind (i.e. that there are gamers who don't care how long it takes), I'm open to think about how Kort could abbreviate this logic of 3 co-occurrences. Currently I'm still looking for a good solution.

**Comment by habi** [comment]
Even though Kort was not designed for OSM *editing*, it's homepage states that 
> Kort helps to improve OpenStreetMap-data.

This is what brought me to download it and use it (also because the OSM editor story on iOS is not so good).

I for my part really like the short-burst gamified approach to improve OSM instead of proper editing, which of course also improves, but can feel more like work.
This also makes *me* want to keep playing with Kort, but I'd still like to see my <del>work</del>game go to OSM in the end.

Maybe - if not confirmed before - the data could be synched to OSM once the monthly high score is not influenced by it anymore, e.g. after a waiting period of 5 weeks or so...

**Comment by sfkeller** [comment]
Thank you all for these interesting and fresh suggestions. @habi mentioned that Kort itself states that it wants to help improve OSM. But that's the side-effect. Of course, I'm also interested that updates takes place after not too long time, thus enouraging those who are aware of that. I just did not yet find another clever way to get a balance with some built-in validation. I definitely want to avoid that users are surprised to see their missions showing up as updates in the web/OSM without warning - like this user https://lists.openstreetmap.org/pipermail/talk/2017-June/078108.html !

**Comment by DaCor1** [comment]
I honestly face-palmed when I saw you guys were still doing this. 

Koins are a quirky byproduct of your game which are valuable to nobody in any context except to see their place on a scoreboard. 

The woeful lack of use of the Kort site before and the Kort app now, when compared to a similar app, StreetComplete, should tell you that this is where you are failing. 

You have pushed less than 1k edits to OSM in how many years? 5?

Time to get rid of this validation

While you're at it, edits (for want of a better word) made in the app should be uploaded under the user's OSM nick

**Comment by sfkeller** [comment]
I first had to read this blog post about the minimally nice maintainer in order to answer your issue politely :-). You really have to be aware that Kort is before a s simple game, not an OSM editor like StreetComplete. Having said this we could lower the co-occurrence parameter to 2. We'll think about.

[1] https://brson.github.io/2017/04/05/minimally-nice-maintainer

**Comment by metaodi** [comment]
I think there is a big misunderstanding here: the goal of Kort was never to maximize the output for OSM, but rather to add good contributions to OSM. So from our point of view, it's far more important to make _correct_ contributions than to make _a lot of_ contributions.

A bit of background info: when we first presented our ideas to the Swiss OSM community (at local events and at a conference back in 2013/2014), we got the feedback that they liked the idea but didn't want to encourage armchair mapping and that we need to ensure a certain level of quality. Basedhon that, we added the radius for which missions are displayed to the user and the validations.

We allow users to login with different accounts (e.g. Google) to have a low barrier of entry. Kort is not for editing OSM, but a fun game to play with the side effect of improving OSM. Hence we use our own bot account to add the data to OSM and do not plan to lose it over false edits.

**Comment by metaodi** [comment]
But a possibility could be to expose all answers via an API, so that others could pick up and submit them to OSM

**Comment by sfkeller** [comment]
Dear all. Thanks for your contributions. 

Kort development has stalled end of 2017 and there are currently not maintainer nor dev resources available on this project. Around October 2018 there will be some shiny activities in order to resurrect this nice game experience. 

Pull requests are still welcome in the meantime.
